vm/src/vm.rs

```diff
@@ -366,6 +366,11 @@ impl<ENV> GVm<ENV> {
                     let v1 = self.get_value(v1);
                     val = self.is_equal_pair(v1, val2)?;
                 }
+                (Value::Nil | Value::Undefined, Value::True) => val = Value::False,
+                (Value::Nil | Value::Undefined, Value::False) => val = Value::True,
```

Comment:
Should undefined be included here? What is it really?

Reply:
I included a description of Undefined in another PR, can discuss further if we need.